### PR TITLE
[nrf fromtree] Ncs lwm2m string api deprecation

### PR DESCRIPTION
**Pull request by jarlamsa**
Bring commits that deprecate the string based LwM2M APIs from upstream zephyr.

### CODE REVIEW

**Comment by rlubos** [comment]
@jarlamsa According to new policy, all `nrf fromtree` commits should reference the upstream commit they were cherrypicked from. Please use `-x` switch wtih cherry-pick to do so.

**Comment by jarlamsa** [comment]
Thanks @rlubos, I made the cherry-picks again with `-x`

**Comment by rlubos** [comment]
@jarlamsa Thanks, although I've just learned that further cherry-picks wrt NCS 2.3 are restricted, therefore this will need approval from release mgmt team before merge.

**Review by SeppoTakalo — APPROVED** [review]

**Review by jheiskan81 — APPROVED** [review]

**Review by juhaylinen — APPROVED** [review]

**Review by desowin — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Review by haakoneh — APPROVED** [review]

**Pull request merged**
